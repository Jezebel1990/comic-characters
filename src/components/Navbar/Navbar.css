.navbar {
    display: flex;
    justify-content: center;
    margin-block: 24px;
    background-color: #fff;
}

.logo {
    width: 320px;
    height: 120px;
}


/* Estilos para dispositivos móveis (máximo de 480px) */
@media screen and (max-width: 480px) {
    .navbar {
        margin-block: 16px; 
        padding: 8px; 
    }

    .logo {
        width: 150px;  
        height: auto; 
    }
}

/* Estilos para tablets (entre 768px e 1024px) */
@media screen and (min-width: 768px) and (max-width: 1024px) {
    .navbar {
        margin-block: 20px;  
        padding: 12px; 
    }

    .logo {
        width: 180px;  
        height: auto; 
    }
}
