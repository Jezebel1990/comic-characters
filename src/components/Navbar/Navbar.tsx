import './Navbar.css'
// @ts-ignore
import logo from '../../assets/logo3x.png'
import React from 'react'

export function Navbar() {
  return (
    <nav className="navbar">
      <img
        src={logo}
        alt="Logo escrito 'Marvel Search heroes', a palavra marvel estÃ¡ em vermelho e o restante em cinza escuro."
        loading="lazy"
        className="logo"
      />
    </nav>
  )
}
