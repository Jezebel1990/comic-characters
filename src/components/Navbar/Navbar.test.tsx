import React from 'react'
import { render, screen } from '@testing-library/react'
import { Navbar } from './Navbar'

describe('<Navbar />', () => {
  it('should render the logo image with correct alt text', () => {
    // Renderiza o componente Navbar
    render(<Navbar />)

    // Verifica se a imagem está sendo renderizada com o alt correto
    const logoImage = screen.getByAltText(
      "Logo escrito 'Marvel Search heroes', a palavra marvel está em vermelho e o restante em cinza escuro."
    )

    expect(logoImage).toBeInTheDocument()
  })

  it('should have the correct loading attribute for the logo image', () => {
    // Renderiza o componente Navbar
    render(<Navbar />)

    // Verifica se a imagem tem o atributo "loading" configurado como "lazy"
    const logoImage = screen.getByAltText(
      "Logo escrito 'Marvel Search heroes', a palavra marvel está em vermelho e o restante em cinza escuro."
    )

    expect(logoImage).toHaveAttribute('loading', 'lazy')
  })

  it('should match the snapshot', () => {
    // Renderiza o componente Navbar
    const { container } = render(<Navbar />)

    // Gera o snapshot
    expect(container.firstChild).toMatchSnapshot()
  })
})
