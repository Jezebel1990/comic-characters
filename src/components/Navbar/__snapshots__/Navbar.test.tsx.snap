// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> should match the snapshot 1`] = `
<nav
  class="navbar"
>
  <img
    alt="Logo escrito 'Marvel Search heroes', a palavra marvel estÃ¡ em vermelho e o restante em cinza escuro."
    class="logo"
    loading="lazy"
    src="test-file-stub"
  />
</nav>
`;
